include/exadg/vector_tools/arbitrary_solution_transfer.h

```diff
@@ -0,0 +1,164 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_VECTOR_TOOLS_ARBITRARY_SOLUTION_TRANSFER_H
+#define INCLUDE_VECTOR_TOOLS_ARBITRARY_SOLUTION_TRANSFER_H
+
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/numerics/vector_tools.h>
+
+namespace ExaDG
+{
+template<int dim>
+class ArbitrarySolutionTransfer
```

Comment:
My last question for now concerns the name of the class. `Arbitrary` might be too general or not specific enough. The name should reflect that the solution is transferred between different grids/triangulations. Hence, I would suggest something like `SolutionTransferBetweenGrids`. 

For clarity, the documentation should mention that the implementation assumes a nodal basis on the dst-side, i.e. the solution evaluated at certain points in space (currently called support points) equals entries of the DoF-vector. The src-side appears to be flexible in terms of nodal vs. modal basis functions (all we use is RPE, which interpolates the src-solution correctly into the dst-points).

Reply:
`SolutionTransferBetweenTriangulations` (as you write in the title) would also be good in my opinion

Reply:
> For clarity, the documentation should mention that the implementation assumes a nodal basis on the dst-side, i.e. the solution evaluated at certain points in space (currently called support points) equals entries of the DoF-vector. The src-side appears to be flexible in terms of nodal vs. modal basis functions (all we use is RPE, which interpolates the src-solution correctly into the dst-points).

If you agree, you could somewhere place this information in the documentation for clarity.

Reply:
As far as I understood the concept of support points, only nodal bases have them. Am I right?

Reply:
Support points are indeed only there for nodal bases. However, other element types have something called "generalized support points", which are some representative points that can be used to convert from values at these points to the values of the weights in the polynomial expansion, be it a modal basis or some mixture (like "moments on the edges"). You can see what is done e.g. in the non-nodal version of the Raviart-Thomas finite element: https://github.com/dealii/dealii/blob/dd99ecfe16df9287195726e4444bb448a5b78fe5/source/fe/fe_raviart_thomas.cc#L679-L707
What you can see here is that there is a "weights" matrix that contains the relevant expansions. For a modal basis (and ignoring the unknowns on faces for Raviart-Thomas at the moment), you can think of this as being the inverse of a Vandermonde matrix.

Reply:
Thanks for clarification :)

Reply:
Ah, so I think I was not aware of "generalized". Against the background provided by @kronbichler , I am now unsure whether the class is really restricted to nodal bases on the dst-side? It looks like we are only working with generalized support points e.g. when collecting the points, and finally we convert the generalized support point values to dof values. If I understand the comment by @kronbichler correctly, the class would/could also work with a modal basis on the dst-side (potentially involving a dense interpolation step with quadratic complexity if this is required by a certain non-nodal basis)?

Reply:
Yes it could work, but not with the current implementation. I have to adapt something for this. Should not take me long.